### PR TITLE
Cut 0.4.0 release

### LINKED ISSUE #86: Cut a new release
Since `v0.3.0` a couple of nice additions have landed (#65, #67 and #70). Maybe its tome for a new release?

### PR DESCRIPTION
**Pull request by jan--f**
This also pulls in #79.

Fixes #86

### CODE REVIEW

**Comment by jan--f** [comment]
cc @squat @simonpasquier

**Comment by squat** [comment]
We technically haven't done this since the 0.1 release of prom-label-proxy, however, would you mind cutting the release/tagging the release in a release-0.4 branch? Also, I would vote against pulling #79 into the master branch in this PR that cuts a release circumventing the approval process in the PR, because I think it makes it hard to track the origin of the code change and is messier

File: main.go [hunk 1]
```diff
@@ -43,12 +43,12 @@ func main() {
...
+	flagset.BoolVar(&enableLabelAPIs, "enable-label-apis", false, "When specified proxy allows to inject label to label APIs like /api/v1/labels and /api/v1/label/<name>/values. "+
+		"NOTE: Enable with care because filtering by matcher is not implemented in older versions of Prometheus/Thanos (see https://github.com/thanos-io/thanos/issues/3351 and https://github.com/prometheus/prometheus/issues/6178). If enabled and "+
+		"any labels endpoint does not support selectors, the injected matcher will have no effect.")
+	flagset.StringVar(&unsafePassthroughPaths, "unsafe-passthrough-paths", "", "Comma delimited allow list of exact HTTP path segments that should be allowed to hit upstream URL without any enforcement. "+
+		"This option is checked after Prometheus APIs, you cannot override enforced API endpoints to be not enforced with this option. Use carefully as it can easily cause a data leak if the provided path is an important "+
+		"API (like /api/v1/configuration) which isn't enforced by prom-label-proxy. NOTE: \"all\" matching paths like \"/\" or \"\" and regex are not allowed.")
```
Comment (simonpasquier) [inline main.go:51]:
```suggestion
		"API (like /api/v1/status/config) which isn't enforced by prom-label-proxy. NOTE: \"all\" matching paths like \"/\" or \"\" and regex are not allowed.")
```

**Comment by jan--f** [comment]
> We technically haven't done this since the 0.1 release of prom-label-proxy, however, would you mind cutting the release/tagging the release in a release-0.4 branch? Also, I would vote against pulling #79 into the master branch in this PR that cuts a release circumventing the approval process in the PR, because I think it makes it hard to track the origin of the code change and is messier

Sure, no problem. I pulled in #79 on @simonpasquier's request. I'll add it to the new PR as well, lets discuss there. If the PR gets approval I'm also happy to rebase.

**Pull request closed without merging**